packages/mobile/src/web3/saga.ts

```diff
@@ -44,12 +44,25 @@ const MNEMONIC_BIT_LENGTH = MnemonicStrength.s256_24words
 export const SYNC_TIMEOUT = 2 * 60 * 1000 // 2 minutes
 const SWITCH_TO_FORNO_TIMEOUT = 15000 // if syncing takes >15 secs, suggest switch to forno
 const WEB3_MONITOR_DELAY = 100
+const BLOCK_AGE_LIMIT = 5 * 60 // if the latest block is older than 5 minutes, the node is not synced.
+
+// Returns true if the block was produced within the block age limit.
+function blockIsFresh(block: Block) {
+  return Math.round(Date.now() / 1000) - Number(block.timestamp) < BLOCK_AGE_LIMIT
+}
+
+enum SyncStatus {
+  UNKNOWN,
+  WAITING,
+  SYNCING,
+}
 
 // checks if web3 claims it is currently syncing and attempts to wait for it to complete
 export function* checkWeb3SyncProgress() {
```

Comment:
One thing to note about `checkWeb3SyncProgress()` is that analytics here may not capture all syncing events, as it is only called when we want to wait for the node to initially sync in onboarding or when the app is first opened (specifically, importing a backup phrase, redeeming an invite, or when the web3Saga is first spawned). It would not capture syncing as someone is using the app after these initializations

For most actions that require the node to be synced, `getConnectedAccount()` is used to wait for geth connectivity, which ensures that geth is connected based on monitoring blocks here: https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/geth/saga.ts#L189

Reply:
I've gone ahead and added some state around the chain head and analytics for events of losing or regaining sync. We could also use this to change from waiting for a connected account to send a transaction, to waiting for a synced account, which may provide better transaction reliability by eliminating cases where we are sending transaction while connected, but not synced. Since we have no idea how common this is, I didn't go that far. I think the analytics will be useful to decide if there is anything to be done here.